ships/0040-build-runtime-class.md

```diff
@@ -0,0 +1,159 @@
+use Kata containers as their runtime so that I can isolate my builds with hardware virtualization.
+
+### Implementation Notes
+
+#### API Update
+
+The `BuildSpec` API for Build and BuildRun will be updated to add the `runtimeClass` field:
+
+```yaml
+spec:
+  ...
+  runtimeClassName: kata # string
+```
+
+This field will correspond to the [runtimeClassName](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#scheduling)
+field on Kubernetes pods.
+
+#### Precedence Ordering and Value Merging
+
+Values in `BuildRun` will override those in the referenced `Build` object (if present). This allows
+the `BuildRun` object to "inherit" values from a parent `Build` object.
+
+#### Impact on Tekton TaskRun
+
+Tekton supports tuning the pod of the `TaskRun` using the
+[podTemplate](https://tekton.dev/docs/pipelines/taskruns/#specifying-a-pod-template) field. When
+Shipwright creates the `TaskRun` for a build, the respective `runtimeClassName` can be passed
+through.
+
+#### Command Line Enhancements
+
+The `shp` CLI _may_ be enhanced to add flags that set the `runtimeClassName` for a `Build` and/or
+`BuildRun`. For example, `shp build run` can have the following new options:
+
+- `--runtime-class=<value>`: this would set the respective `spec.runtimeClassName` value on the
+  generated `BuildRun`.
+
+### Test Plan
+
+- Unit testing can verify that the generated `TaskRun` object for a build contains the desired pod
+  template fields.
+- End to end testing may prove challenging, as KinD is not designed to support multiple container
+  runtimes. Implementations like Kata Containers are designed for real clusters on cloud provider
+  infrastructure.
+
+
+### Release Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Removing a deprecated feature [if necessary]
+
+Not applicable.
+
+#### Upgrade Strategy [if necessary]
+
+The `runtimeClassName` field will be optional and default to Golang empty values.
+On upgrade, these values will remain empty on existing `Build`/`BuildRun` objects.
+
+### Risks and Mitigations
+
+#### Pod Scheduling Collision
+
+The `RuntimeClass` object (set and maintained by cluster admins/platform teams) also has options
+to set defaults for [pod scheduling](https://kubernetes.io/docs/concepts/containers/runtime-class/#scheduling),
+such as node selectors and tolerations. As of Kubernetes 1.32, this is a "beta" feature enabled
+on most Kubernetes distributions by default. These scheduler options are merged on pod admission;
+there is a risk that pods are rejected and builds fail due to colliding pod scheduler values.
+
+The reconcile loop for `BuildRun` could in theory catch this situation ahead of time and fail the
+build prior to pod creation. We could also test this scenario and see how Tekton `TaskRuns` behave;
+if the `TaskRun` fails with a reasonable error message, then Shipwright `BuildRun`s can simply echo
+the information in their status.
```

Comment:
I prefer if we continue to create the TaskRun without checking anything else. We already don't validate other things in that area that could go wrong. We for example do not check the pod security enforcement on the namespace against the security context of the build strategy.

Reply:
For the existence check I tend to say it is not a requirement but would be nice if it fails build registration.